### PR TITLE
tldr 1.3.0 (new formula)

### PR DESCRIPTION
**Pull request by danielcompton**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [ ] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

The formula doesn't pass audit because the tldr-cpp-client doesn't have enough users.

```
$ brew audit --new-formula tldr
tldr:
  * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
Error: 1 problem in 1 formula
```

A similar PR was rejected in https://github.com/Homebrew/legacy-homebrew/pull/47867 because it doesn't have enough users. At first glance, this Formula would seem to have the same problem. 

However the main repo which has the users, contributions, and attention is https://github.com/tldr-pages/tldr. From that repo, you can install an NPM plugin which provides tldr results. However that has obvious issues, the biggest for me is that npm is not homebrew. That is the reasoning for adding this formula instead.

This formula was taken from https://github.com/tldr-pages/homebrew-tldr.

### CODE REVIEW

**Comment by apjanke** [comment]
The rationale for allowing this formula despite being "non-notable" makes sense to me.

File: Formula/tldr.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  desc "Simplified and community-driven man pages"
+  homepage "https://tldr-pages.github.io"
+
+  url "https://github.com/tldr-pages/tldr-cpp-client/archive/v1.2.0.tar.gz"
+  sha256 "db2e70b3f4209c67b5b9896e052bf3d9e77570dcb04d0c9932776aa44cca9d3e"
+
+  head "https://github.com/tldr-pages/tldr-cpp-client.git"
+
+  depends_on "curl"
+  depends_on "libzip" => :build
```
Comment (Leandros) [inline Formula/tldr.rb]: you have to set that as a build dependency, too.

Comment (apjanke) [inline Formula/tldr.rb]: Do you mean as a run time dependency?

Comment (zmwangx) [inline Formula/tldr.rb]:
```
$ brew linkage tldr
System libraries:
  /usr/lib/libSystem.B.dylib
Homebrew libraries:
  /usr/local/opt/curl/lib/libcurl.4.dylib (curl)
  /usr/local/opt/libzip/lib/libzip.4.dylib (libzip)
```

So yes, should be a runtime dependency too.

**Comment by danielcompton** [comment]
Thanks for the helpful feedback all, I've updated the formula to @zmwangx's suggestion.

**Comment by zmwangx** [comment]
Thanks for fixing it!

@Leandros Do you feel strongly about the `"DEBUG=-march=native -O2"` flags to `make`?

**Comment by Leandros** [comment]
@zmwangx No, but since it's currently a none-bottled formula, they won't do any harm. I'll look into creating a binary which can then be distributed via brew in the next days.

**Comment by zmwangx** [comment]
> I'll look into creating a binary which can then be distributed via brew in the next days.

Our CI bottles formulae automatically so once we merge this we'll have binaries already.

**Comment by Leandros** [comment]
Oh, than we should remove `-march=native`, it might created problems.

**Comment by zmwangx** [comment]
@Leandros Currently the flags aren't included in this PR. If you don't think we should add them then we're golden.

**Comment by Leandros** [comment]
They, unfortunately, are: https://github.com/tldr-pages/tldr-cpp-client/blob/master/Makefile#L7
However, we can simply change this in the formula, instead of the main repository (which might be a little bit dirty, though), by adding `make CFLAGS=` as build command.

**Comment by zmwangx** [comment]
Oh I see. I'm personally fine with adding some flags, but will see if other maintainers have a problem with that.

File: Formula/tldr.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  url "https://github.com/tldr-pages/tldr-cpp-client/archive/v1.2.0.tar.gz"
+  sha256 "db2e70b3f4209c67b5b9896e052bf3d9e77570dcb04d0c9932776aa44cca9d3e"
+  head "https://github.com/tldr-pages/tldr-cpp-client.git"
+
+  depends_on "curl"
+  depends_on "libzip"
+  depends_on "pkg-config" => :build
+
+  def install
+    system "make"
```
Comment (zmwangx) [inline Formula/tldr.rb]: According to @Leandros, this should better be

```ruby
system "make", "CFLAGS="
```

@danielcompton Could you please make the change? Thanks.

Comment (danielcompton) [inline Formula/tldr.rb]: Done

Comment (Leandros) [inline Formula/tldr.rb]: Alternatively, or probably even better, would be 

    system "make", "DEBUG=-O2"

This will enable some, safe, optimizations, and disable the debugging flags.

**Comment by Leandros** [comment]
Sorry for being a little late :/
I think it would be better to add:

     system "make", "DEBUG=-O2"

**Comment by zmwangx** [comment]
@Leandros Which do you think is safer long term? People don't tend to change the build instructions in future updates unless they break.

From my initial interactions with `tldr` it doesn't seem to be a performance-critical or otherwise slow program, so I don't feel strongly about `-O2` either way.

**Comment by ilovezfs** [comment]
The CFLAGS= shouldn't be necessary. If you call `make` specifically, superenv refurbishes the args

**Comment by Leandros** [comment]
No, tldr is not performance critical. It's either I/O or network which will slow it down. However, `-O2` only enables safe optimizations, which shouldn't be critical or brake in a future update.

**Comment by ilovezfs** [comment]
Bottling defaults to -march=core2 and -Os

**Comment by zmwangx** [comment]
Sorry for the misinformation...

**Comment by Leandros** [comment]
@ilovezfs That's sweet. If you call `make CFLAGS=-march="-march=core2 -Os` while bottling we're safe now.

File: Formula/tldr.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  url "https://github.com/tldr-pages/tldr-cpp-client/archive/v1.2.0.tar.gz"
+  sha256 "db2e70b3f4209c67b5b9896e052bf3d9e77570dcb04d0c9932776aa44cca9d3e"
+  head "https://github.com/tldr-pages/tldr-cpp-client.git"
+
+  depends_on "curl"
+  depends_on "libzip"
+  depends_on "pkg-config" => :build
+
+  def install
+    system "make", "DEBUG=-O2"
```
Comment (zmwangx) [inline Formula/tldr.rb]: `DBUEG=-O2` should be removed here because it won't actually do anything

Comment (Leandros) [inline Formula/tldr.rb]: Sorry for the confusion. :worried:

Comment (zmwangx) [inline Formula/tldr.rb]: I should be the one saying that to you...

**Comment by danielcompton** [comment]
@Leandros I've given you push access to my fork, might be simplest for you to make the last changes :)

**Comment by zmwangx** [comment]
Well I can update that for you too if you don't mind other people touching your code.

**Comment by zmwangx** [comment]
Fixed. CI should be happy other than the "not notable enough" audit problem. Let's sit back and wait for a few 👍 from other maintainers.

**Comment by Leandros** [comment]
Formula looks good. What is meant with "not stable enough"?

**Comment by zmwangx** [comment]
```
$ brew audit --new-formula tldr
tldr:
  * GitHub repository not notable enough (<20 forks, <20 watchers and <50 stars)
Error: 1 problem in 1 formula
```

But as explained in the top post, the actual project is https://github.com/tldr-pages/tldr which is definitely notable enough.

**Comment by Leandros** [comment]
Ahh, sorry, I read "stable" instead of "notable". 😊

**Comment by Leandros** [comment]
I've also updated our tap, to reflect the latest changes, including those suggested by you! Thanks a lot! 👍 

https://github.com/tldr-pages/homebrew-tldr/commit/60102af4188e2ae4affaada8d946e64cf257946a

Do we want to include 1.3.0 in the update?

**Comment by zmwangx** [comment]
> Do we want to include 1.3.0 in the update?

We certainly do. Shipping 1.2.0 only to immediately update to 1.3.0 would be pretty bad.

> I've also updated our tap

Once we merge this you can sunset the tap, point your users here, and submit any future changes here.

**Comment by Leandros** [comment]
> Once we merge this you can sunset the tap, point your users to core, and submit any future changes here.

Yes, will do! 👍

File: Formula/tldr.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class Tldr < Formula
+  desc "Simplified and community-driven man pages"
+  homepage "https://tldr-pages.github.io"
+  url "https://github.com/tldr-pages/tldr-cpp-client/archive/v1.3.0.tar.gz"
+  sha256 "6210ece3f5d8f8e55b404e2f6c84be50bfdde9f0d194a271bce751a3ed6141be"
+  head "https://github.com/tldr-pages/tldr-cpp-client.git"
+
+  depends_on "curl"
```
Comment (DomT4) [inline Formula/tldr.rb]: Why can't this package use the system `curl`?

File: Formula/tldr.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  depends_on "libzip"
+  depends_on "pkg-config" => :build
```
Comment (DomT4) [inline Formula/tldr.rb]: Nit, but you'll make me especially happy if you put `:build` dependencies first in the deps list.

File: Formula/tldr.rb (hunk 4 above)
Comment (Leandros) [inline Formula/tldr.rb]: Is it guaranteed to be decently up to date? I remember there were incompatibilities.

**Comment by zmwangx** [comment]
More thoughts folks, especially regarding eligibility? If not, I think we can merge this in a day or two.

File: Formula/tldr.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  homepage "https://tldr-pages.github.io"
+  url "https://github.com/tldr-pages/tldr-cpp-client/archive/v1.3.0.tar.gz"
+  sha256 "6210ece3f5d8f8e55b404e2f6c84be50bfdde9f0d194a271bce751a3ed6141be"
+  head "https://github.com/tldr-pages/tldr-cpp-client.git"
+
+  depends_on "pkg-config" => :build
+  depends_on "libzip"
+
+  def install
+    system "make"
```
Comment (zmwangx) [inline Formula/tldr.rb]: I took a look at `Makefile` and the `install` target already depends on `all`, so this line is redundant. I'll push a fix.

**Comment by zmwangx** [comment]
I said "a day or two" in a comment made two days ago, so there we go.

**Comment by zmwangx** [comment]
Merged as 3b3786d. Thank you @danielcompton for your contribution to Homebrew! And thank you @Leandros for your help. You can now point your users to core!

**Comment by Leandros** [comment]
Awesome! Thank you very much.
